concert_service_manager/src/concert_service_manager/service_cache_manager.py

```diff
@@ -0,0 +1,468 @@
+            else:
+                self._loginfo("load from cache: [%s]" % service_configuration_file)
+        return (check_result, service_configuration_file)
+
+    def _create_service_cache(self):
+        """
+          Create cache as loading service configuration from default value.
+
+        """
+        # read resource file
+        loaded_profiles = {}
+        service_configurations = load_solution_configuration_from_default(rocon_python_utils.ros.find_resource_from_string(self._resource_name))
+        for service_configuration in service_configurations:
+            resource_name = rocon_python_utils.ros.check_extension_name(service_configuration['resource_name'], '.service')
+            overrides = service_configuration['overrides']
+            try:
+                loaded_profile = self._load_service_profile_from_default(resource_name, overrides)
+                self._save_service_profile(loaded_profile)
+                loaded_profiles[loaded_profile['name']] = copy.deepcopy(loaded_profile)
+            except rospkg.ResourceNotFound as e:
+                self.logwarn('Cannot load service configuration: [%s]' % resource_name)
+                continue
+        # save solution configuration
+        self._save_solution_configuration(loaded_profiles)
+
+    def _load_service_profile_from_default(self, resource_name, overrides):
+        """
+          Load service profile information from default.
+
+          :param resource_name: default resource name. It is composed package name/services name.
+          :type resource_name: str
+          :param overrides: overrided informantion. Its propertise are 'name', 'description', 'icon', 'priority', 'interactions' and 'parameters'. If the property does not setting, its has default value as None
+          :type overrides: dict
+
+          :return: dictionary data about service profile
+          :rtype: dict
+
+          :raises: :exc:`rospkg.ResourceNotFound` 
+
+        """
+        # load service profile from default
+        try:
+            file_name = rocon_python_utils.ros.find_resource_from_string(resource_name)
+            with open(file_name) as f:
+                loaded_profile = yaml.load(f)
+        except rospkg.ResourceNotFound as e:
+            raise e
+        loaded_profile['resource_name'] = resource_name
+
+        # set priority to default if it was not configured
+        if 'priority' not in loaded_profile.keys():
+            loaded_profile['priority'] = scheduler_msgs.Request.DEFAULT_PRIORITY
+        for key in loaded_profile:
+            if key in overrides and overrides[key] is not None:
+                loaded_profile[key] = overrides[key]
+        if 'launcher_type' not in loaded_profile.keys():  # not set
+            loaded_profile['launcher_type'] = concert_msgs.ServiceProfile.TYPE_SHADOW
+        loaded_profile['name'] = loaded_profile['name'].strip().lower().replace(" ", "_")
```

Comment:
This also can use https://github.com/robotics-in-concert/rocon_tools/pull/73/files#diff-f384ea7500354bbc728b8b10d20f119bR38